pkg/server/ingest/server.go

```diff
@@ -151,11 +151,20 @@ func (hs *ingestionServerCfg) Run() (err error) {
 
 	// run fasthttp server
 	var g run.Group
-	g.Add(func() error {
-		return s.Serve(hs.ln)
-	}, func(e error) {
-		_ = hs.ln.Close()
-	})
+
+	if config.IsTlsEnabled() && config.GetTLSCertificatePath() != "" && config.GetTLSPrivateKeyPath() != "" {
```

Comment:
we do not need the second check for the certPath here.
During bootup, if tls is enabled and the certpath is not specified, then we should exit the process and log an error with appropriate message. 
This will avoid doing extra checks on every request that we receive